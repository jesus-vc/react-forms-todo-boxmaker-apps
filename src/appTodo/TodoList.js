import NewTodoForm from "./NewTodoForm";
import { useState } from "react";
import Todo from "./Todo";

const TodoList = () => {
  const [tasks, setTasks] = useState([]);

  const addTask = (newTask) => {
    setTasks((tasks) => [...tasks, newTask]);
  };

  const removeTask = (id) => {
    setTasks(tasks.filter((task) => task.id !== id));
  };

  const todoComponents = tasks.map(({ id, newTask }) => (
    <Todo key={id} task={newTask} removeTask={() => removeTask(id)}></Todo>
  ));

  return (
    <div>
      <ul>{todoComponents}</ul>
      <NewTodoForm addTask={addTask}></NewTodoForm>
    </div>
  );
};

export default TodoList;
