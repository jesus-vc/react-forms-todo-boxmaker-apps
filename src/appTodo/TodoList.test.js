import { render, screen, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

// smoke test
it("renders without crashing", () => {
  render(<TodoList />);
});

//snapshot text
it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders a new component matching the text input", () => {
  render(<TodoList />);

  const textInput = screen.getByLabelText("New Task:");
  const btn = screen.queryByText("Add New Task!");
  fireEvent.change(textInput, { target: { value: "Wash cars." } });

  fireEvent.click(btn);

  const newTask = screen.getByText("Wash cars.");

  expect(newTask).toBeInTheDocument();
});

it("removes a task when button is clicked", () => {
  // create new task
  render(<TodoList />);
  const textInput = screen.getByLabelText("New Task:");
  const btn = screen.queryByText("Add New Task!");
  fireEvent.change(textInput, { target: { value: "Do something dangerous!" } });
  fireEvent.click(btn);
  const newTask = screen.getByText("Do something dangerous!");
  expect(newTask).toBeInTheDocument();

  // delete the new task
  const btnDelete = screen.queryByText("X");
  fireEvent.click(btnDelete);
  expect(newTask).not.toBeInTheDocument();
});
