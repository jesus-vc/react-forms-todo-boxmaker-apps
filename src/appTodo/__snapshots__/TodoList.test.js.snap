// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <ul />
    <div>
      <form>
        <div>
          <label
            for="newTask"
          >
            New Task:
          </label>
          <br />
          <input
            id="newTask"
            name="newTask"
            type="text"
            value=""
          />
        </div>
        <button
          id="newTaskButton"
        >
          Add New Task!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
